libminifi/src/RemoteProcessorGroupPort.cpp

```diff
@@ -264,20 +264,27 @@ void RemoteProcessorGroupPort::refreshRemoteSite2SiteInfo() {
     if (!response_body.empty()) {
       std::string controller = std::string(response_body.begin(), response_body.end());
       logger_->log_debug("controller config %s", controller.c_str());
-      Json::Value value;
-      Json::Reader reader;
-      bool parsingSuccessful = reader.parse(controller, value);
-      if (parsingSuccessful && !value.empty()) {
-        Json::Value controllerValue = value["controller"];
-        if (!controllerValue.empty()) {
-          Json::Value port = controllerValue["remoteSiteListeningPort"];
-          if (client_type_ == sitetosite::CLIENT_TYPE::RAW && !port.empty())
-            this->site2site_port_ = port.asInt();
+
```

Comment:
If we're going to change the JSON code we should decouple it from this class.

Reply:
All I did here was translate the jsoncpp code to rapidjson. I didn't test it.